src/GameState/index.tsx

```diff
@@ -206,6 +207,67 @@ export default function GameState({ navigation, route, player }: any) {
     };
   }, []);
 
+  const setMarbleClicked = async (marbleClicked: {
+    x: number;
+    y: number;
+    value: number;
+    isExit: boolean;
+  }) => {
+    try {
+      const response = await fetch(
+        `${API_URL}/games/${gameId}/marble-clicked`,
+        {
+          method: 'PUT',
+          body: JSON.stringify({ marbleClicked }),
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: 'Bearer ' + player.token,
+          },
+        },
+      );
+      if (response.status >= 200 && response.status < 300) {
+        dispatch({ type: 'game', value: { ...state.game, marbleClicked } });
+      }
+    } catch (error) {
+      toast.show({
+        title: 'Error',
+        status: 'error',
+        description: 'Cannot select this marble',
+      });
+    }
+  };
+
+  async function checkAndMoveMarble(direction: string) {
+    try {
+      if (await isMovePossible(gameId, player.token, player.id, direction)) {
+        if (state.game?.marbleClicked) {
+          const coordinates = {
+            x: state.game.marbleClicked.x,
+            y: state.game.marbleClicked.y,
+          };
+          const playerForAPI = getPlayerObject(player);
+          const response = await moveMarble(
+            gameId,
+            coordinates,
+            playerForAPI,
+            direction,
+            player.token,
+          );
+
+          const game = (await response.json()) as Game;
```

Comment:
You never test de status of the response ... and you dont know if you have a game in your response .... Missing to much check points ... :(